pygfx/helpers/_gizmo.py

```diff
@@ -493,10 +493,10 @@ def _update_visibility(self):
 
     # %% Event handling
 
-    def add_default_event_handlers(self, renderer, camera):
+    def add_default_event_handlers(self, viewport, camera):
         # Store objects that we need outside the event handling.
-        self._renderer = renderer
-        self._canvas = renderer.target
+        viewport = Viewport.from_viewport_or_renderer(viewport)
+        self._viewport = viewport
```

Comment:
this is a slight discrepancy between the gizmo and the controls, why does one need to store a reference on self but not the other?

Reply:
Because the gizmo also does a bunch of things at draw time, where it needs the viewport and rect. Will add a comment to explain.